docs-ref-conceptual/dotnet-howto-migrate-app-service.md

```diff
@@ -1,54 +1,108 @@
+
+## Database size  
+For SQL Server databases, please check the current [SQL Database pricing](http://azure.microsoft.com/pricing/details/sql-database).  
+
+## Azure Active Directory (AAD) integration  
+AAD does not work with free apps. To use AAD, you must upgrade the app SKU. See [App Service pricing](https://azure.microsoft.com/pricing/details/app-service/windows/).
 
 ## Monitoring and Diagnostics
 Your current on-premises solutions for monitoring and diagnostics are unlikely to work in the cloud. However, Azure provides tools for logging, monitoring, and diagnostics so that you can identify and debug issues with web apps. You can easily enable diagnostics for your web app in its configuration, and you can view the logs recorded in Azure Application Insights. [Learn more about enabling diagnostics logging for web apps](https://docs.microsoft.com/azure/app-service/web-sites-enable-diagnostic-log).
 
 ## Connection Strings and application settings
-One option for keeping information safe is to use [Azure KeyVault](https://docs.microsoft.com/azure/key-vault/), a service that securely stores sensitive information used in your application. Alternatively, you can store this data as an App Service setting.
+Consider using [Azure KeyVault](https://docs.microsoft.com/azure/key-vault/), a service that securely stores sensitive information used in your application. Alternatively, you can store this data as an App Service setting.
 
 ## DNS
-You may need to update DNS configurations based on the requirements of your application. These DNS settings can be configured in the App Service [custom domain settings](https://docs.microsoft.com/azure/app-service/app-service-web-tutorial-custom-domain). Another factor to consider is [binding an existing custom SSL certificate](https://docs.microsoft.com/azure/app-service/app-service-web-tutorial-custom-ssl).
+You may need to update DNS configurations based on the requirements of your application. These DNS settings can be configured in the App Service [custom domain settings](https://docs.microsoft.com/azure/app-service/app-service-web-tutorial-custom-domain). 
+
+## Azure App Service with Windows Containers
+If your app cannot be migrated directly to App Service, consider App Service using Windows Containers (%TODO%: Link), which enables usage of the GAC, COM components, MSIs, full access to .NET FX APIs, DirectX, and more.
 
-## File system and Storage
-If your application persists data, you will need to update it to use Azure Storage instead. Azure Storage is a service that provides file shares for sharing via SMB protocol, blob storage, simple queues, and non-relational tables. [Learn more about Azure Storage file shares](https://docs.microsoft.com/azure/storage/files/storage-files-introduction).
+## Additional Reading
+
```

Comment:
There's more detail on App Service environment limitations at [projectkudu/kudu](https://github.com/projectkudu/kudu/wiki/Azure-Web-App-sandbox), might be worth linking here.